### PR TITLE
chore(doc): add node lease controller description in architecture page

### PR DESCRIPTION
**Pull request by caozhuozi**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:

the architecture page of the document is missing the introduction of the _Node Lease Controller_.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Hi reviewers,
I just found the architecture page of the document is missing the introduction of the _Node Lease Controller_.
Hope I understand the controller correctly.
I also noticed the _Stage Controller_ in the code. But I'm not sure if this should be documented (maybe the user might not care about this internal controller?) Please also let me know if it needs to be documented. I will also add a description for it then. Thanks!

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: site/content/en/docs/design/architecture.md [hunk 1]
```diff
@@ -18,7 +18,9 @@ It can run in any environment, use in-cluster or kubeconfig to connect to a `kub
 
 So far, `kwok` has implemented the following controllers:
 
-- Node Controller - It is responsible for selecting the node to simulate, and then simulating the node's lifecycle, just by updating the heartbeat of the node.
+- Node Controller - It is responsible for selecting the nodes to simulate, and then simulating the nodes' lifecycle, just by updating the node status field (which originally should be reported by kubelet).
+  You can also define and customize node's lifecycle through [Stages Configuration].
+- Node Lease Controller - It is responsible for reporting node heartbeats by creating and renewing the node lease objects for those managed nodes. See [Node HeartBeats] and [KEP 589] for more details.
```
Comment (wzshiming) [inline site/content/en/docs/design/architecture.md:23]: nit: it's a whole word

```suggestion
- Node Lease Controller - It is responsible for reporting node heartbeats by creating and renewing the node lease objects for those managed nodes. See [Node Heartbeats] and [KEP 589] for more details.
```

File: site/content/en/docs/design/architecture.md [hunk 2]
```diff
@@ -18,7 +18,9 @@ It can run in any environment, use in-cluster or kubeconfig to connect to a `kub
 
 So far, `kwok` has implemented the following controllers:
 
-- Node Controller - It is responsible for selecting the node to simulate, and then simulating the node's lifecycle, just by updating the heartbeat of the node.
+- Node Controller - It is responsible for selecting the nodes to simulate, and then simulating the nodes' lifecycle, just by updating the node status field (which originally should be reported by kubelet).
+  You can also define and customize node's lifecycle through [Stages Configuration].
```
Comment (wzshiming) [inline site/content/en/docs/design/architecture.md:22]: The Node Controller and the Pod Controller parts are actually the same, and so is the new [Stage Controller](
https://github.com/kubernetes-sigs/kwok/blob/main/pkg/kwok/controllers/stage_controller.go), which I think we can describe uniformly.

Comment (caozhuozi) [reply site/content/en/docs/design/architecture.md:22]: sure.  Thanks, Shiming. I haven't read through the code of the stage controller in detail yet. I will finish it and find a way to describe them. Will discuss it with you in the following comments.

Comment (caozhuozi) [reply site/content/en/docs/design/architecture.md:22]: Hi @wzshiming, I tried a new description that includes the new _stage controller_, and grouped the pod controller, node controller, together with stage controller into a category. Please feel free to give comments and suggestions.

So far, `kwok` has implemented 2 types of controllers:

- Resource Lifecycle Simulation Controller - This type of controller is used to simulate the lifecycle of Kubernetes resources. You can define and customize a resource's lifecycle through [Stages Configuration].
  Specifically, `kwok` has the following controllers of this type:
  * Node Controller - It is responsible for selecting the nodes to simulate, and then simulating the nodes' lifecycle, just by updating the node status field (which originally should be reported by kubelet).
  * Pod Controller - It is responsible for simulating the lifecycle of pods scheduled to those selected nodes by updating the pod status field (which originally should be reported by kubelet).
  * Stage Controller -  It is later introduced for a more general purpose, aiming to simulate the lifecycle of other Kubernetes resource types besides `v1.Node` and `v1.Pod`.
- Node Lease Controller - It is a dedicated controller for reporting node heartbeats by creating and renewing the node lease objects for those managed nodes. See [Node Heartbeats] and [KEP 589] for more details.

Comment (wzshiming) [reply site/content/en/docs/design/architecture.md:22]: The overall feeling is good, it is worth noting that the Resource Lifecycle Simulation Controller currently does not exist, and the Node Controller and the Pod Controller are equivalent to the specialized implementation of the Stage Controller.

Comment (caozhuozi) [reply site/content/en/docs/design/architecture.md:22]: Thanks, Shiming. Let me add some more clarification based on your explanations.

Comment (caozhuozi) [reply site/content/en/docs/design/architecture.md:22]: Hi @wzshiming, I added more clarification based on your explanations. Please feel free to give comments and suggestions.


So far, `kwok` has 2 types of controllers:

- Resource Lifecycle Simulation Controller - This type of controller is used to simulate the lifecycle of Kubernetes resources. You can define and customize a resource's lifecycle through [Stages Configuration].
  It is worth noting that Resource Lifecycle Simulation Controller is just a conceptual term for document purpose and does not really exist in `kwok`. Specifically, `kwok` has implemented the following controllers of this type:
  * Node Controller - It is responsible for selecting the nodes to simulate, and then simulating the nodes' lifecycle, just by updating the node status field (which originally should be reported by kubelet).
  * Pod Controller - It is responsible for simulating the lifecycle of pods scheduled to those selected nodes by updating the pod status field (which originally should be reported by kubelet).
  * Stage Controller -  It is later introduced for a more general purpose, aiming to simulate the lifecycle of other Kubernetes resource types besides `v1.Node` and `v1.Pod`.
    You can think the Node Controller and the Pod Controller are the specialized implementation of the Stage Controller.
- Node Lease Controller - It is a dedicated controller in `kwok` for reporting node heartbeats by creating and renewing the node lease objects for those managed nodes. See [Node Heartbeats] and [KEP 589] for more details.

Comment (wzshiming) [reply site/content/en/docs/design/architecture.md:22]: It's looking good so far.

**Review by windsonsea — COMMENTED** [review]
Some text suggestions to make two sentences clear

File: site/content/en/docs/design/architecture.md [hunk 3]
```diff
@@ -16,10 +16,15 @@ This document will introduce the architecture of KWOK.
 
 It can run in any environment, use in-cluster or kubeconfig to connect to a `kube-apiserver` of cluster, and then manage the resources of the cluster.
 
-So far, `kwok` has implemented the following controllers:
+So far, `kwok` has 2 types of controllers:
```
[the remark this answers is not in the document]
Comment (caozhuozi) [reply site/content/en/docs/design/architecture.md:19]: Thanks for helping improve the writing. @windsonsea

File: site/content/en/docs/design/architecture.md [hunk 4]
```diff
@@ -16,10 +16,15 @@ This document will introduce the architecture of KWOK.
...
 
-- Node Controller - It is responsible for selecting the node to simulate, and then simulating the node's lifecycle, just by updating the heartbeat of the node.
-- Pod Controller - It is responsible for pod that is on selected node, and plays the stage of pod's lifecycle.
+- Resource Lifecycle Simulation Controller - This type of controller is used to simulate the lifecycle of Kubernetes resources. You can define and customize a resource's lifecycle through [Stages Configuration].
+  It is worth noting that Resource Lifecycle Simulation Controller is just a conceptual term for document purpose and does not really exist in `kwok`. Specifically, `kwok` has implemented the following controllers of this type:
```
[the remark this answers is not in the document]
Comment (caozhuozi) [reply site/content/en/docs/design/architecture.md:22]: sure！

**Review by windsonsea — COMMENTED** [review]
/lgtm

**Pull request merged**